### PR TITLE
[ISSUE-3769][test] Deepen ApacheAclReadServiceTest with dedup, master-skip, root-cause and empty-table coverage

### PR DESCRIPTION
**Pull request by yyqdbngt**
### Motivation

The existing `ApacheAclReadServiceTest` covers the partial-failure aggregation and null-row tolerance, but leaves the iteration guards of the broker-table walk unverified: duplicate master addresses, broker entries with no master, the root-cause unwrapping for failure messages, and a missing broker table.

### Changes

- `skipsDuplicateMasterAddressAcrossBrokerEntries`: two broker entries sharing one master address trigger a single `listAcl` call and a single policy key.
- `skipsBrokerEntriesWithoutMasterAddress`: entries with no address table or only slave addresses never reach `listAcl` and yield an empty, non-partial result.
- `usesDeepestRootCauseAsFailureMessage`: a wrapped failure records the deepest cause message instead of the outer one.
- `treatsMissingBrokerTableAsEmptyResult`: a `ClusterInfo` with no broker table short-circuits to an empty result without any `listAcl` invocation.

### Verification

```
mvn -B test -Dtest=ApacheAclReadServiceTest
[INFO] Tests run: 7, Failures: 0, Errors: 0, Skipped: 0
[INFO] BUILD SUCCESS
```

### CODE REVIEW

**Review by RockteMQ-AI — APPROVED** [review]
### Summary
Test-only PR adding deeper coverage for existing test classes. The new test cases cover meaningful edge cases (dedup, no-op, boundary conditions, failure passthrough) with clean assertions. LGTM.

---
*Automated review by github-manager-bot*

File: server/src/test/java/org/apache/rocketmq/studio/instance/acl/ApacheAclReadServiceTest.java [hunk 1]
```diff
@@ -123,4 +125,74 @@ private void executeWith(RuntimeAdminClientResolver resolver, MQAdminExt admin)
...
+        assertThat(result.getFailuresByBroker()).isEmpty();
+        assertThat(result.isPartial()).isFalse();
+    }
+
+    @Test
+    void usesDeepestRootCauseAsFailureMessage() throws Exception {
+        RuntimeAdminClientResolver resolver = mock(RuntimeAdminClientResolver.class);
+        MQAdminExt admin = mock(MQAdminExt.class);
+        when(admin.examineBrokerClusterInfo()).thenReturn(clusterInfo("broker-a:10911"));
+        when(admin.listAcl("broker-a:10911", null, null))
```
Comment (utafrali) [inline server/src/test/java/org/apache/rocketmq/studio/instance/acl/ApacheAclReadServiceTest.java:175]: The exception chain here is only two levels deep (`IllegalStateException -> RuntimeException`), so a simple `getCause()` would pass this test just as easily as a true root-cause traversal. To actually verify the algorithm walks to the bottom, use a three-level chain:

```java
.thenThrow(new IllegalStateException("outer",
    new RuntimeException("middle",
        new RuntimeException("root cause"))));
```

Then the assertion `containsEntry("broker-a:10911", "root cause")` would fail if the implementation stops at `getCause()` instead of traversing to the deepest cause.

File: server/src/test/java/org/apache/rocketmq/studio/instance/acl/ApacheAclReadServiceTest.java [hunk 2]
```diff
@@ -123,4 +125,74 @@ private void executeWith(RuntimeAdminClientResolver resolver, MQAdminExt admin)
...
+                .thenThrow(new IllegalStateException("outer", new RuntimeException("root cause")));
+        executeWith(resolver, admin);
+
+        RemoteAclReadResult result = new ApacheAclReadService(resolver).listRules("instance-1", null, null);
+
```
Comment (utafrali) [inline server/src/test/java/org/apache/rocketmq/studio/instance/acl/ApacheAclReadServiceTest.java:180]: This test records a broker failure but never checks `result.isPartial()`. Every other test that has a meaningful result state (tests 2 and 4) asserts on `isPartial()`. Since a failure should mark the result partial, add:

```java
assertThat(result.isPartial()).isTrue();
```

File: server/src/test/java/org/apache/rocketmq/studio/instance/acl/ApacheAclReadServiceTest.java [hunk 3]
```diff
@@ -123,4 +125,74 @@ private void executeWith(RuntimeAdminClientResolver resolver, MQAdminExt admin)
...
+        ClusterInfo clusterInfo = new ClusterInfo();
+        clusterInfo.setBrokerAddrTable(Map.of("first", first, "second", second));
+        when(admin.examineBrokerClusterInfo()).thenReturn(clusterInfo);
+        when(admin.listAcl("broker-a:10911", null, null)).thenReturn(List.of());
+        executeWith(resolver, admin);
+
+        RemoteAclReadResult result = new ApacheAclReadService(resolver).listRules("instance-1", null, null);
+
+        verify(admin).listAcl("broker-a:10911", null, null);
+        assertThat(result.getPoliciesByBroker()).containsOnlyKeys("broker-a:10911");
```
Comment (utafrali) [inline server/src/test/java/org/apache/rocketmq/studio/instance/acl/ApacheAclReadServiceTest.java:146]: `skipsDuplicateMasterAddressAcrossBrokerEntries` is missing an `isPartial()` assertion. Tests 2 and 4 both check `assertThat(result.isPartial()).isFalse()` after successful/empty outcomes. Add the same here for consistency and to document the expected state:

```java
assertThat(result.isPartial()).isFalse();
```

**Review by utafrali — COMMENTED** [review]
The four new tests cover real gaps in the broker-table walk and are structurally sound, but `usesDeepestRootCauseAsFailureMessage` uses a two-level exception chain that a naive `getCause()` would also satisfy, and that test is missing the `isPartial()` assertion present in comparable tests.
